src/seedu/addressbook/AddressBook.java

```diff
@@ -467,32 +479,53 @@ private static String getMessageForPersonsDisplayedSummary(ArrayList<String[]> p
     }
 
     /**
-     * Extracts keywords from the command arguments given for the find persons command.
+     * Extracts keywords from the command arguments given for the find persons
+     * command.
      *
-     * @param findPersonCommandArgs full command args string for the find persons command
+     * @param findPersonCommandArgs full command args string for the find
+     *            persons command
      * @return set of keywords as specified by args
      */
     private static Set<String> extractKeywordsFromFindPersonArgs(String findPersonCommandArgs) {
         return new HashSet<>(splitByWhitespace(findPersonCommandArgs.trim()));
     }
 
     /**
-     * Retrieves all persons in the full model whose names contain some of the specified keywords.
+     * Retrieves all persons in the full model whose names contain some of the
+     * specified keywords.
      *
      * @param keywords for searching
-     * @return list of persons in full model with name containing some of the keywords
+     * @return list of persons in full model with name containing some of the
+     *         keywords
      */
     private static ArrayList<String[]> getPersonsWithNameContainingAnyKeyword(Collection<String> keywords) {
         final ArrayList<String[]> matchedPersons = new ArrayList<>();
         for (String[] person : getAllPersonsInAddressBook()) {
             final Set<String> wordsInName = new HashSet<>(splitByWhitespace(getNameFromPerson(person)));
-            if (!Collections.disjoint(wordsInName, keywords)) {
-                matchedPersons.add(person);
-            }
+            findMatchCaseInsensitive(keywords, matchedPersons, person, wordsInName);
+
+      
         }
         return matchedPersons;
     }
 
+    private static void findMatchCaseInsensitive(Collection<String> keywords, final ArrayList<String[]> matchedPersons,
```

Comment:
notice the arrow head like code indentation; it is one of the indications that SLAP may not be so good in the code.
try to improve it!